File: smartsocket/ss68/java/tutorial/lesson4/send3.java
/*
 * Copyright (c) 1997-2006 TIBCO Software Inc.
 * All rights reserved.
 * For more information, please contact:
 * TIBCO Software Inc., Palo Alto, California, USA
 *
 * $Id: //dev/packages/examples/ss/dev/68/src/java/tutorial/lesson4c/send3.java#1 $
 */

//----------------------------------------------------------------------
// send3.java -- write an INFO and then NUMERIC_DATA messages

import java.io.*;
import com.smartsockets.*;

public class send3 {

  public static void main(String[] argv) {
    try {
      Tut.setOption("ss.project", "smartsockets");

      TipcSrv srv=TipcSvc.getSrv();
      if (!srv.create()) {
        Tut.exitFailure("Couldn't connect to RTserver!\n");
      } // if

      // send a message of type INFO
      TipcMsg msgi = TipcSvc.createMsg(TipcMt.INFO);
      msgi.setDest("/ss/tutorial/lesson4");
      msgi.appendStr("Hello World!");
      srv.send(msgi);
      srv.flush();

      // create a message of type NUMERIC_DATA
      TipcMsg msg = TipcSvc.createMsg(TipcMt.NUMERIC_DATA);

      // set the destination subject of the message
      msg.setDest("/ss/tutorial/lesson4");

      // each time through the loop send a NUMERIC_DATA
      // message with three values
      for (int i = 0; i < 30; i = i + 3) {
        msg.setNumFields(0);
        msg.appendStr("X");
        msg.appendReal8(i);
        msg.appendStr("Y");
        msg.appendReal8(i+1.0);
        msg.appendStr("Z");
        msg.appendReal8(i+2.0);

        // send the message
        srv.send(msg);
        srv.flush();
      }

      // disconnect from RTserver
      srv.destroy();
    } catch (TipcException e) {
      Tut.warning(e);
    } // catch
  } // main

} // send3

File: smartsocket/ss68/examples/smrtsock/tutorial/lesson4/send3.c
/*
 * Copyright (c) 1991-2006 TIBCO Software Inc.
 * All rights reserved.
 * For more information, please contact:
 * TIBCO Software Inc., Palo Alto, California, USA
 *
 * $Id: //dev/packages/examples/ss/dev/68/src/c/tutorial/lesson4/send3.c#1 $
 */

/* send3.c - send a INFO and then a series of NUMERIC_DATA messages */
/* $RTHOME/examples/smrtsock/tutorial/lesson4/send3.c */

#include <rtworks/ipc.h>

int main(int argc, char **argv)
{
  T_OPTION option;
  T_IPC_MSG msg;
  T_INT4 i;

  /* Set the name of the project */
  option = TutOptionLookup("project");
  TutOptionSetEnum(option, "smartsockets");

  /* Connect to RTserver */
  if (!TipcSrvCreate(T_IPC_SRV_CONN_FULL)) {
    TutOut("Could not connect to RTserver!\n");
    TutExit(T_EXIT_FAILURE);
  } /* if */

  /* Send an INFO message */
  TipcSrvMsgWrite("/tutorial/lesson4", 
		  TipcMtLookupByNum(T_MT_INFO), TRUE,
		  T_IPC_FT_STR, "Hello World!", NULL);

  /* Create a message of type NUMERIC_DATA */
  msg = TipcMsgCreate( TipcMtLookupByNum(T_MT_NUMERIC_DATA) );

  /* Set the destination subject of the message */
  TipcMsgSetDest(msg, "/tutorial/lesson4");

  /* Each time through the loop send a NUMERIC_DATA message with 3 values */
  for (i = 0; i < 30; i = i + 3) {
    /* 
     * In order to re-use the message, reset the number of the 
     * fields in the message to zero 
     */
    TipcMsgSetNumFields(msg, 0);

    /* Build the message with 3 variable-value pairs */
    TipcMsgAppendStrReal8(msg, "X", (T_REAL8)i);
    TipcMsgAppendStrReal8(msg, "Y", (T_REAL8)i+1);
    TipcMsgAppendStrReal8(msg, "Z", (T_REAL8)i+2);

    /* Send the message */
    TipcSrvMsgSend(msg, TRUE);
    TipcSrvFlush();
  } /* for */

  /* Destroy the message */
  TipcMsgDestroy(msg);
	
  /* Disconnect from RTserver */
  TipcSrvDestroy(T_IPC_SRV_CONN_NONE);

} /* main */
